File: csharp/isogram/Isogram.cs
using System;
using System.Linq;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var sortedString = word.ToLower().ToCharArray();
        Array.Sort(sortedString);

        bool isoGram = true;
        for (int i = 0; i < sortedString.Length - 1; i++)
        {
            if (char.IsLetter(sortedString[i]) && sortedString[i] == sortedString[i + 1])
            {
                isoGram = false;
                break;
            }
        }
        return isoGram;
    }
}

File: go/isogram/isogram.go
package isogram

import (
	"regexp"
	"sort"
	"strings"
)

func IsIsogram(word string) bool {
	if len(word) == 1 {
		return true
	}

	regex := regexp.MustCompile(`\w`)
	onlyWord := regex.FindAllString(word, -1)
	sort.Strings(onlyWord)

	for i := 0; i < len(onlyWord)-1; i++ {
		if strings.ToLower(onlyWord[i]) == strings.ToLower(onlyWord[i+1]) {
			return false
		}
	}
	return true
}
